Eggstravaganza/Assets/Scripts/EggSpawn.cs

```diff
@@ -0,0 +1,97 @@
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+using Random = UnityEngine.Random;
+
+[Serializable]
+public class Egg
+{
+    public GameObject prefab;
+    [Range(0f, 100)]
+    public float chance = 100;
+    [HideInInspector]
+    public double weight;
+}
+
+public class EggSpawn : MonoBehaviour
+{
+    public List<Egg> eggs;
```

Comment:
```suggestion
    [SerializeField]
    private List<Egg> m_Eggs;
```
for now perhaps? And can expose as a property later if needed?

Reply:
whoops you're right, done!